expression/builtin_string.go

```diff
@@ -3019,6 +3019,54 @@ func (b *builtinFormatSig) evalString(row chunk.Row) (string, bool, error) {
 	if isNull || err != nil {
 		return "", isNull, err
 	}
+	if strings.Contains(x, ".") {
```

Comment:
Please wrap up this code snippet into a single function, and call it in `evalNumDecArgsForFormat`, because `builtinFormatWithLocaleSig::evalString` need this round up as well.